server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/node/StarTreeNodeType.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex.datacube.startree.node;
+
+/**
+ * Represents the different types of nodes in a StarTree data structure.
+ *
+ * <p>
+ * In order to handle different node types, we use a byte value to represent the node type.
+ * This enum provides a convenient way to map byte values to their corresponding node types.
+ *
+ * <p>
+ * Star and Null Nodes are represented as special cases. Default is the general case.
+ * Star and null nodes are represented with negative ordinal values to ensure that they are
+ * sorted before the default nodes, which are sorted based on their dimension values.
+ *
+ * <p>
+ * The node type can be one of the following:
+ * <ul>
+ *     <li>Star Node: Represented by the value -2. A star node is a special node that represents
+ *     all possible values for a dimension.</li>
+ *     <li>Null Node: Represented by the value -1. A null node indicates the absence of any value
+ *     for a dimension.</li>
+ *     <li>Default Node: Represented by the value 0. A default node represents a node with an
+ *     actual dimension value.</li>
+ * </ul>
+ *
+ * By default, we want to consider nodes as default node.
+ *
+ * @opensearch.experimental
+ * @see StarTreeNode
+ */
+public enum StarTreeNodeType {
+
+    /**
+     * Represents a star node type.
+     *
+     */
+    STAR("star", (byte) -2),
+
+    /**
+     * Represents a null node type.
+     */
+    NULL("null", (byte) -1),
+
+    /**
+     * Represents a default node type.
+     */
+    DEFAULT("default", (byte) 0);
```

Comment:
I think the idea is any node created by default, should be a default node. 
Since we want the Star Node to be the first node, and null node to be a second node. -2, -1, and 0 helps with that.
Hope that answers your question.